compiler/test/Dialect/Linalg/fuse-attention.mlir

```diff
@@ -160,3 +160,71 @@ transform.sequence failures(propagate) {
   %0 = transform.structured.match attributes {__root__} in %arg0 : (!pdl.operation) -> !pdl.operation
   %transformed, %loops:2 = transform.structured.fuse_ext %0 {tile_interchange = [0, 1, 4, 3, 2], tile_sizes = [2, 4, 0, 0, 0]}
 }
+
+// -----
+
+// support split-head multi-query-attention, see https://arxiv.org/pdf/1911.02150.pdf
```

Comment:
From my understanding, in MQA, only Q has heads, so your K/V get recomputed in this case?

Reply:
The tile and fuse logic is inherently greedy, which can lead to recomputation of operations. To prevent Q/V from being fused into the loop, I have explicitly marked them to stop here.